plugins/jade/src/test/scala/org/broadinstitute/monster/sbt/BigQueryMetadataGeneratorSpec.scala

```diff
@@ -0,0 +1,80 @@
+package org.broadinstitute.monster.sbt
+
+import org.broadinstitute.monster.sbt.model._
+import org.broadinstitute.monster.sbt.model.bigquery.BigQueryColumn
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class BigQueryMetadataGeneratorSpec extends AnyFlatSpec with Matchers {
+
+  private val exampleTable = MonsterTable(
+    name = new JadeIdentifier("file"),
+    columns = Vector(
+      SimpleColumn(
+        name = new JadeIdentifier("sample_id"),
+        datatype = DataType.String,
+        `type` = ColumnType.PrimaryKey,
+        links = Vector(
+          Link(
+            tableName = new JadeIdentifier("sample"),
+            columnName = new JadeIdentifier("id")
+          )
+        )
+      ),
+      SimpleColumn(
+        name = new JadeIdentifier("file_type"),
+        datatype = DataType.String,
+        `type` = ColumnType.PrimaryKey
+      ),
+      SimpleColumn(
+        name = new JadeIdentifier("data_type"),
+        datatype = DataType.String,
+        `type` = ColumnType.Required
+      ),
+      SimpleColumn(
+        name = new JadeIdentifier("creation_date"),
+        datatype = DataType.Date
+      ),
+      SimpleColumn(
+        name = new JadeIdentifier("size"),
+        datatype = DataType.Integer
+      )
+    ),
+    structColumns = Vector(
+      StructColumn(
+        name = new JadeIdentifier("metrics"),
+        structName = new JadeIdentifier("file_metrics"),
+        `type` = ColumnType.Required
+      ),
+      StructColumn(
+        name = new JadeIdentifier("comments"),
+        structName = new JadeIdentifier("comments"),
+        `type` = ColumnType.Repeated
+      )
+    )
+  )
+
+  behavior of "BigQueryMetadataGenerator"
```

Comment:
We're just checking to see that the metadata tables got created properly in BQ, right? Not looking for contents within the schemas?

Reply:
This test is checking that the contents of the generated schemas are correct. Nothing is actually created in BQ by the code in the plugin; it just generates files that are useful for interacting with BQ.